dev/service-registry/play-integration-javadsl/src/main/scala/com/lightbend/lagom/play/LagomPlayModule.scala

```diff
@@ -4,58 +4,85 @@
 package com.lightbend.lagom.play
 
 import java.net.URI
+import java.util.{ List => JList }
 import java.util.Optional
-
-import scala.collection.JavaConverters._
-import scala.compat.java8.FutureConverters._
-import com.google.inject.Provider
-import com.lightbend.lagom.javadsl.api.ServiceAcl
-import com.lightbend.lagom.javadsl.api.ServiceInfo
-import com.lightbend.lagom.javadsl.api.transport.Method
 import javax.inject.Inject
-import javax.inject.Singleton
 
 import com.lightbend.lagom.internal.javadsl.registry.{ ServiceRegistry, ServiceRegistryService }
-import play.api.Configuration
-import play.api.Environment
-import play.api.inject.ApplicationLifecycle
-import play.api.inject.Binding
-import play.api.inject.Module
+import com.lightbend.lagom.javadsl.api.{ ServiceAcl, ServiceInfo }
+import com.lightbend.lagom.javadsl.api.transport.Method
+import com.typesafe.config.Config
+import play.api.inject.{ ApplicationLifecycle, Binding, Module }
+import play.api.{ Configuration, Environment, Logger }
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+import scala.util.{ Success, Try }
 
 class LagomPlayModule extends Module {
-  override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = {
+  private val logger = Logger(this.getClass)
+
+  override def bindings(environment: Environment, config: Configuration): Seq[Binding[_]] = {
+    val maybeServiceInfoBinding: Option[Binding[ServiceInfo]] = prepareServiceInfoBinding(config)
+    val playRegistry = bind[PlayRegisterWithServiceRegistry].toSelf.eagerly()
+
     Seq(
-      bind[PlayRegisterWithServiceRegistry].toSelf.eagerly(),
-      bind[ServiceInfo].toProvider[PlayServiceInfoProvider]
-    )
+      playRegistry
+    ) ++ maybeServiceInfoBinding.toList
+
   }
-}
 
-@Singleton
-class PlayServiceInfoProvider @Inject() (config: Configuration) extends Provider[ServiceInfo] {
-  lazy val get = {
-    val serviceName = config.underlying.getString("lagom.play.service-name")
-    new ServiceInfo(serviceName)
+  private def prepareServiceInfoBinding(config: Configuration) = {
+    val triedServiceName = Try(config.underlying.getString("lagom.play.service-name"))
+    val triedAcls: Try[JList[_ <: Config]] = Try(config.underlying.getConfigList("lagom.play.acls"))
```

Comment:
If the value is explicitly null (rather than simply missing) or a wrong type, meaning that the keys were set incorrectly, wouldn't we want to propagate that error rather than swallowing it and falling back on different behaviour? It implies that the user is trying to set those configurations, but made a mistake.